### PR TITLE
Mark complaints upvoted by user, Add boolean fields for upvotes, subscription

### PR DESCRIPTION
**Pull request by preetamozarde3**
Boolean fields for checking whether the user has upvoted or been subscribed to the complaint have been added to the Venter model. The previous code has been changed to reflect the same. Also, the redundant fields vote_count and complaintSubscribed have been refactored.
Also, a feature has been added to mark the complaints that have been upvoted by a user.

### CODE REVIEW

File: app/src/main/java/app/insti/adapter/ComplaintsAdapter.java [hunk 1]
```diff
@@ -297,9 +297,9 @@ public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int po
         List<User> userList = complaintList.get(position).getUsersUpVoted();
         for (User user : userList) {
             if (user.getUserID().equals(userID))
-               complaintList.get(position).setVoteCount(1);
+               complaintList.get(position).setComplaintUpvoted(true);
             else
```
Comment (pulsejet) [inline app/src/main/java/app/insti/adapter/ComplaintsAdapter.java]: IIRC you no longer should need to do this since the server would send the value of `is_subscribed`

Comment (preetamozarde3) [reply app/src/main/java/app/insti/adapter/ComplaintsAdapter.java]: My bad no it shouldn't. I only removed it from the ComplaintFragment and forgot about this. I have fixed it though and added a feature to mark the complaints the users have upvoted in the ComplaintsAdapter.

File: app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java [hunk 2]
```diff
@@ -47,8 +47,8 @@ public void onStart() {
         swipeContainer.post(new Runnable() {
             @Override
             public void run() {
+                callServerToGetNearbyComplaints();
                 if (complaints == null) {
-                    callServerToGetNearbyComplaints();
```
Comment (pulsejet) [inline app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java]: Why this change?

File: app/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -3,6 +3,7 @@
     <color name="colorPrimary">#536dfe</color>
     <color name="colorPrimaryDark">#0043ca</color>
     <color name="colorAccent">#ffd740</color>
+    <color name="colorBlue">#0000ff</color>
```
Comment (pulsejet) [inline app/src/main/res/values/colors.xml]: This _cannot_ fit into the theme. Use colorPrimary instead if you want a blue.

File: app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java (hunk 2 above)
Comment (preetamozarde3) [reply app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java]: Once you navigate to a complaint details fragment from the complaints home or me page, on the back press the complaints are not updated. So if a user upvotes or comments on the complaint it won't be reflected on the back press.
So this change refreshes the complaints every time the page is loaded.

File: app/src/main/res/values/colors.xml (hunk 3 above)
Comment (preetamozarde3) [reply app/src/main/res/values/colors.xml]: I've made the change and pushed it

File: app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java (hunk 2 above)
Comment (pulsejet) [reply app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java]: This change was made deliberately in 6f6736df814818a294ea64b306135711f449b730 since loading complaints every time the user navigates back to complaints is:
1) Excessive unnecessary server calls. Especially with paging, if the user has loaded 50 complaints, then this is effectively 5 unnecessary expensive API calls if the user ever does scroll down to the complaint (it doesn't matter anyway if the user doesn't)
2) Terrible UX, for obvious reasons

There are two ways around this:
1) Use a global static memcache (similar to `UpdatableList`s in Utils for events, bodies, notifications etc.) and update this as things change.
2) Don't fix it, since the tradeoff of UX is too much comparable to the negligible gain that we are getting here.

Comment (preetamozarde3) [reply app/src/main/java/app/insti/fragment/ComplaintsHomeFragment.java]: Alright I've reverted the change

**Comment by pulsejet** [comment]
Merged @preetamozarde3
